src/SendGrid/Helpers/Mail/SendGridMessage.cs

```diff
@@ -243,15 +252,15 @@ public void AddTos(List<EmailAddress> emails, int personalizationIndex = 0, Pers
                     this.Personalizations[personalizationIndex].Tos = new List<EmailAddress>();
                 }
 
-                this.Personalizations[personalizationIndex].Tos.AddRange(emails);
+                this.Personalizations[personalizationIndex].Tos.AddRange(emails.Except(this.Personalizations[personalizationIndex].Tos));
```

Comment:
Is this effectively the same as `this.Personalizations[personalizationIndex].Tos.AddRange(emails).Distinct()`?

Reply:
Using Distinct() would use both properties. Assuming the Email property is the "key" property, this does a "Distinct" across the Email property.

Reply:
No. That code would still add duplicates within the `emails` list. This code takes all `emails` from the list that don't intersect with the current list and adds the non-duplicates to the list.